drivers/wifi/esp32/Kconfig.esp32wifi

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (c) 2019 Manulytica Ltd
+# Copyright (c) 2018 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Kconfig - ESP32 AT WiFi driver options
+
+menuconfig WIFI_ESP32
+	bool "ESP32 WiFi driver support"
+	default n
+	depends on WIFI
+	select NET_SOCKETS
+	select WIFI_OFFLOAD
+	select NET_L2_WIFI_MGMT
+	select MODEM
+	select MODEM_RECEIVER
```

Comment:
I would rather not add new drivers which use `MODEM_RECEIVER`.  Can we use `MODEM_CONTEXT` instead?